crates/apollo-router/src/configuration/mod.rs

```diff
@@ -317,42 +312,43 @@ mod tests {
         assert_config_snapshot!("testdata/config_opentelemetry_jaeger_full.yml");
     }
 
-    #[cfg(any(feature = "otlp-tonic", feature = "otlp-grpcio", feature = "otlp-http"))]
+    #[cfg(any(feature = "otlp-grpc", feature = "otlp-http"))]
     #[test]
     fn ensure_configuration_api_does_not_change_common() {
         // NOTE: don't take a snapshot here because the optional fields appear with ~ and they vary
         // per implementation
+
+        #[cfg(feature = "otlp-http")]
+        serde_yaml::from_str::<Configuration>(include_str!(
+            "testdata/config_opentelemetry_otlp_tracing_http_common.yml"
+        ))
+        .unwrap();
+
+        #[cfg(feature = "otlp-grpc")]
         serde_yaml::from_str::<Configuration>(include_str!(
-            "testdata/config_opentelemetry_otlp_tracing_common.yml"
+            "testdata/config_opentelemetry_otlp_tracing_tonic_common.yml"
         ))
         .unwrap();
     }
 
-    #[cfg(feature = "otlp-tonic")]
+    #[cfg(feature = "otlp-grpc")]
     #[test]
     fn ensure_configuration_api_does_not_change_tonic() {
         assert_config_snapshot!("testdata/config_opentelemetry_otlp_tracing_tonic_basic.yml");
         assert_config_snapshot!("testdata/config_opentelemetry_otlp_tracing_tonic_full.yml");
     }
 
-    #[cfg(feature = "otlp-grpcio")]
-    #[test]
-    fn ensure_configuration_api_does_not_change_grpcio() {
-        assert_config_snapshot!("testdata/config_opentelemetry_otlp_tracing_grpcio_basic.yml");
-        assert_config_snapshot!("testdata/config_opentelemetry_otlp_tracing_grpcio_full.yml");
-    }
-
     #[cfg(feature = "otlp-http")]
     #[test]
     fn ensure_configuration_api_does_not_change_http() {
         assert_config_snapshot!("testdata/config_opentelemetry_otlp_tracing_http_basic.yml");
         assert_config_snapshot!("testdata/config_opentelemetry_otlp_tracing_http_full.yml");
     }
 
-    #[cfg(all(feature = "tls", feature = "otlp-tonic"))]
+    #[cfg(all(feature = "tls", feature = "otlp-grpc"))]
     #[test]
     fn ensure_configuration_api_does_not_change_tls_config() {
-        assert_config_snapshot!("testdata/config_opentelemetry_otlp_tls.yml");
+        assert_config_snapshot!("testdata/config_opentelemetry_otlp_tracing_tonic_tls.yml");
```

Comment:
ditto, maybe we can refer to it as grpc and not tonic now that there's no ambiguity anymore